includes/core/DataTransferObjects/class-data_transfer_object.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace WCPay\Core\DataTransferObjects;
+
+use ReflectionClass;
+use ReflectionException;
+
+abstract class Data_Transfer_Object
+{
+	/**
+	 * Return array of all properties.
+	 *
+	 * @return array
+	 */
+	public function to_array() {
```

Comment:
I'll probably come back to this, but why do we need a `to_array()` method? Can't we implement `ArrayAccess` here?

Reply:
For simplicity, I used `to_array()`. We could also use `ArrayAccess`. Please let me know what you think that it would be best. If you ask me, I would leave `to_array` since it can be easily edited based on the needs.